src/transaction/log_checkpoint_info.cpp

```diff
@@ -0,0 +1,159 @@
+
+#include "log_checkpoint_info.hpp"
+namespace cublog
+{
+
+  void checkpoint_info::load_trantable_snapshot ()
+  {
+  }
+
+  int
+  log_lsa_size (LOG_LSA log, cubpacking::packer &serializator, std::size_t start_offset, std::size_t size_arg)
+  {
+    size_t size = size_arg;
+    size += serializator.get_packed_bigint_size (start_offset + size);
+    size += serializator.get_packed_short_size (start_offset + size);
+
+    return size;
+  }
+
+  void
+  log_lsa_pack (LOG_LSA log, cubpacking::packer &serializator)
+  {
+    serializator.pack_bigint (log.pageid);
+    serializator.pack_short (log.offset);
+  }
+
+  LOG_LSA
+  log_lsa_unpack (cubpacking::unpacker &deserializator)
+  {
+    int64_t pageid;
+    short offset;
+    deserializator.unpack_bigint (pageid);
+    deserializator.unpack_short  (offset);
+
+    return log_lsa (pageid, offset);
+  }
+
+  void
+  checkpoint_info::pack (cubpacking::packer &serializator) const
+  {
+
+    log_lsa_pack (m_start_redo_lsa, serializator);
+    log_lsa_pack (m_snapshot_lsa, serializator);
+
+    serializator.pack_bigint (m_trans.size ());
+    for (const checkpoint_tran_info tran_info : m_trans)
+      {
+	serializator.pack_int (tran_info.isloose_end);
+	serializator.pack_int (tran_info.trid);
+	serializator.pack_int (tran_info.state);
+	log_lsa_pack (tran_info.head_lsa, serializator);
+	log_lsa_pack (tran_info.tail_lsa, serializator);
+	log_lsa_pack (tran_info.undo_nxlsa, serializator);
+
+	log_lsa_pack (tran_info.posp_nxlsa, serializator);
+	log_lsa_pack (tran_info.savept_lsa, serializator);
+	log_lsa_pack (tran_info.tail_topresult_lsa, serializator);
+	log_lsa_pack (tran_info.start_postpone_lsa, serializator);
+	serializator.pack_c_string (tran_info.user_name, strlen (tran_info.user_name));
+      }
+
+    serializator.pack_bigint (m_sysops.size ());
+    for (const checkpoint_sysop_info sysop_info : m_sysops)
+      {
+	serializator.pack_int (sysop_info.trid);
+	log_lsa_pack (sysop_info.sysop_start_postpone_lsa, serializator);
+	log_lsa_pack (sysop_info.atomic_sysop_start_lsa, serializator);
+      }
+
+    serializator.pack_bool (m_has_2pc);
+  }
+
+  void
+  checkpoint_info::unpack (cubpacking::unpacker &deserializator)
+  {
+    m_start_redo_lsa = log_lsa_unpack (deserializator);
+    m_snapshot_lsa = log_lsa_unpack (deserializator);
+
+    size_t m_trans_size = 0;
+    deserializator.unpack_bigint (m_trans_size);
+    for (int i = 0; i < m_trans_size; i++)
+      {
+	LOG_INFO_CHKPT_TRANS chkpt_trans;
+
+	deserializator.unpack_int (chkpt_trans.isloose_end);
+	deserializator.unpack_int (chkpt_trans.trid);
+
+	int tran_state_int;
+	deserializator.unpack_int (tran_state_int);
+	chkpt_trans.state = static_cast<TRAN_STATE> (tran_state_int);
```

Comment:
Use unpacker function unpack_from_int that automatically casts the into to the required type.
```suggestion
	deserializator.unpack_from_int (chkpt_trans.state);
```